pkg/solana/chainreader/chain_reader.go

```diff
@@ -300,14 +375,42 @@ func (s *SolanaChainReaderService) addAccountRead(namespace string, genericName
 		inputAccountIDLDef = codec.NilIdlTypeDefTy
 		reader = newAccountReadBinding(namespace, genericName, "", false)
 	}
-	if err := s.addCodecDef(true, namespace, genericName, codec.ChainConfigTypeAccountDef, idl, inputAccountIDLDef, readDefinition.InputModifications); err != nil {
+	if err := s.addCodecDef(true, namespace, genericName, idl, inputAccountIDLDef, readDefinition.InputModifications); err != nil {
 		return err
 	}
+
 	s.bindings.AddReadBinding(namespace, genericName, reader)
 
 	return nil
 }
 
+func (s *ContractReaderService) addEventRead(
+	namespace, genericName string,
+	contractAddress solana.PublicKey,
+	_ codec.IDL,
+	_ codec.IdlEvent,
+	readDefinition config.ReadDefinition,
+	events EventsReader,
+) error {
+	subKeys := [][]string{}
+	for _, onChain := range readDefinition.IndexedFields {
+		subKeys = append(subKeys, strings.Split(onChain, "."))
+	}
+
+	filter := toLPFilter(readDefinition.PollingFilter, contractAddress, subKeys)
+
+	s.filters = append(s.filters, filter)
+	s.bindings.AddReadBinding(namespace, genericName, newEventReadBinding(
+		namespace,
+		genericName,
+		readDefinition.IndexedFields,
```

Comment:
The event binding won't need to know the onchain names.  Those are only used to decode the logs coming in from the rpc server.  (In other words, the purpose of the onChain field names is only to tell `Process` which fields in the struct it should extract and index).   Neither `QueryKey()` nor anything it calls needs to know the onchain field names, since the values have already been extracted and indexed.

So here, we could just pass the ordered slice of onChain names:
```suggestion
		onChainNames
```
where
```
var onChainNames []string
// build by iterating over IndexedFields
```
Although if you prefer to just pass the whole readDefinition.IndexedFields, and let the `AddReadBinding` method extract the part it needs that works too.

Ultimately, the most convenient form to save it in the `eventReadBinding` struct will be as a reverse map from onChainName to subKeyIndex:
```
var subKeyIndexes [string]int
```

Reply:
> Although if you prefer to just pass the whole readDefinition.IndexedFields, and let the AddReadBinding method extract the part it needs that works too.

This might actually be better, come to think of it... it would eliminate the need for iterating over this list an extra time, just to extract the offChain names, since you have to iterate over them anyway in `AddReadBinding`